orchagent/fgnhgorch.cpp

```diff
@@ -1396,23 +1534,63 @@ bool FgNhgOrch::doTaskFgNhgMember(const KeyOpFieldsValuesTuple & t)
             /* skip addition if next-hop already exists */
             if (fgNhg_entry->second.next_hops.find(next_hop) != fgNhg_entry->second.next_hops.end())
             {
-                SWSS_LOG_INFO("FG_NHG member %s already exists, skip", next_hop.to_string().c_str());
+                SWSS_LOG_INFO("FG_NHG member %s already exists for %s, skip",
+                        next_hop.to_string().c_str(), fg_nhg_name.c_str());
                 return true;
             }
-            fgNhg_entry->second.next_hops[next_hop] = bank;
+            FGNextHopInfo fg_nh_info = {};
+            fg_nh_info.bank = bank;
+
+            if (!link.empty())
+            {
+                /* Identify link oper state for initialization */
+                Port p;
+                if (!gPortsOrch->getPort(link, p))
+                {
+                    SWSS_LOG_WARN("FG_NHG member %s added to %s with non-existent link %s, link mapping skipped",
+                            next_hop.to_string().c_str(), fg_nhg_name.c_str(), link.c_str());
+                }
+                else
+                {
+                    link_oper = LINK_DOWN; /* Default operational state is down */
+                    fg_nh_info.link = link;
+                    if (p.m_oper_status == SAI_PORT_OPER_STATUS_UP)
+                    {
+                        link_oper = LINK_UP;
+                    }
+                    auto link_info = fgNhg_entry->second.links.find(link);
+                    fg_nh_info.link_oper_state = link_oper;
+
+                    if (link_info != fgNhg_entry->second.links.end())
+                    {
+                        link_info->second.push_back(next_hop);
+                    }
+                    else
+                    {
+                        std::vector<IpAddress> ips;
+                        ips.push_back(next_hop);
+                        fgNhg_entry->second.links[link] = ips;
+                    }
+                    SWSS_LOG_INFO("Added link %s to ip %s map", link.c_str(), key.c_str());
+                }
+            }
 
             /* query and check the next hop is valid in neighOrcch */
             if (!m_neighOrch->hasNextHop(nhk))
             {
                 SWSS_LOG_INFO("Nexthop %s is not resolved yet", nhk.to_string().c_str());
-                return true;
             }
-
-            /* add next-hop into SAI group */
-            if (!validNextHopInNextHopGroup(nhk))
+            else if (link_oper)
```

Comment:
is the flow here handled for the case where port doesn't exists (get_port failed)?

Reply:
If get_port fails then link_oper will have the default value of 1 per: https://github.com/Azure/sonic-swss/blob/9a586c6cdbc95e2b75e429964f19047d68c9b751/orchagent/fgnhgorch.cpp#L1498

So it will end up going into the else if and call into validNextHopInNextHopGroup(assuming hasNextHop is true).